File: back_end/src/main/java/com/example/entity/request/Repair.java
package com.example.entity.request;

import com.example.entity.response.CustomerResp;
import com.example.entity.response.DeptResp;
import com.example.entity.response.EmpResp;
import com.example.entity.response.ProductResp;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.format.annotation.DateTimeFormat;

import java.util.Date;

/**
 * @author: liuleimin
 * @date: 2020/10/29
 * @description:
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Repair {
    private Integer repairId;
    private Integer empId;//接单人
    @DateTimeFormat(pattern = "yyyy-MM-dd  HH:mm:ss")
    private Date repairDate;//接单时间
    private String repairProblem;//zhut
    private String repairPersonnel;
    private String repairAppointment;
    private String repairActual;
    private String repairGdstate;
    private String repairFault;
    private String repairHfjl;
    private Integer repairWxfy;
    private String repairSfzb;
    private String repairLxr;
    private String repairsjhm;
    private Integer cusId;
    private Integer deptId;
    private Integer productId;
    private Integer orderId;
    private Integer dictId;
    private Integer pageNum;
    private Integer pageSize;
    private CustomerResp customerResp;
    private EmpResp empResp;
    private DeptResp deptResp ;
    private ProductResp productResp;
    private SzOrder szOrder;
    private String cusName;
    private String empName;
}

File: front_end/src/network/system/repair.js
import {postJson,get} from "../request";

export const repairHttp = {
  add:function (repair) {
    let params = {
      url:'/repair/add',
      data:{
        'repairProblem':repair.repairProblem,
        'repairPersonnel':repair.repairPersonnel,
        'repairDate':repair.repairDate,
        'repairAppointment':repair.repairAppointment,
        'repairActual':repair.repairActual,
        'repairGdstae':repair.repairGdstae,
        'repairFault':repair.repairFault,
        'repairHfjl':repair.repairHfjl,
        'repairWxfy':repair.repairWxfy,
        'repairSfzb':repair.repairSfzb,
        'repairLxr':repair.repairLxr,
        'repairsjhm':repair.repairsjhm,
        'cusId':repair.cusId,
        'deptId':repair.deptId,
        'empId':repair.empId
      }
    }
    return postJson(params)
  },
  get:function (repairId) {
    let params = {
      url:'/repair',
      data:{
        'repairId':repairId
      }
    }
    return get(params)
  },
  update:function (repair) {
    let params = {
      url:'/repair/update',
      data:{
        'repairId':repair.repairId,
        'repairProblem':repair.repairProblem,
        'repairPersonnel':repair.repairPersonnel,
        'repairDate':repair.repairDate,
        'repairAppointment':repair.repairAppointment,
        'repairActual':repair.repairActual,
        'repairGdstae':repair.repairGdstae,
        'repairFault':repair.repairFault,
        'repairHfjl':repair.repairHfjl,
        'repairWxfy':repair.repairWxfy,
        'repairSfzb':repair.repairSfzb,
        'repairLxr':repair.repairLxr,
        'repairsjhm':repair.repairsjhm,
        'cusId':repair.cusId,
        'deptId':repair.deptId,
        'empId':repair.empId
      }
    }
    return postJson(params)
  },
  del:function (repairId) {
    let params = {
      url:'/repair/del',
      data:{
        'repairId':repairId
      }
    }
    return postJson(params)
  },
  list:function (repair) {
    let params = {
      url:'/repair/list',
      data:{
        'repairProblem':repair.repairProblem,
        'repairPersonnel':repair.repairPersonnel,
        'repairDate':repair.repairDate,
        'repairAppointment':repair.repairAppointment,
        'repairActual':repair.repairActual,
        'repairGdstae':repair.repairGdstae,
        'cusId':repair.cusId,
        'deptId':repair.deptId,
        'empId':repair.empId

      }
    }
    return get(params)
  },
  queryEmp:function (repair) {
    let params = {
      url:'/repair/list',
      data:{
        'empId':repair.empId,
        'repairProblem':repair.repairProblem,
        'repairPersonnel':repair.repairPersonnel,
        'repairAppointment':repair.repairAppointment,
        'repairActual':repair.repairActual,
        'repairGdstae':repair.repairGdstae,
        'repairFault':repair.repairFault,
        'repairHfjl':repair.repairHfjl,
        'repairWxfy':repair.repairWxfy,
        'repairSfzb':repair.repairSfzb,
        'cusId':repair.cusId,
        'deptId':repair.deptId,
        'productId':repair.productId,
        'orderId':repair.orderId,
        'dictId':repair.dictId,
        'pageNum':repair.pageNum,
        'pageSize':repair.pageSize
      }
    }
    return get(params)
  },
  listPage:function (pageNum,pageSize) {
    let params = {
      url:'/repair/list',
      data:{
        'pageNum':pageNum,
        'pageSize':pageSize
      }
    }
    return get(params)
  }
}
